### PR TITLE
Iroha client query

### PR DESCRIPTION
**Pull request by humb1t**
Signed-off-by: Nikita Puzankov <humb1t@yandex.ru>

### Description of the Change

API for making Iroha queries from TCP client.

### CODE REVIEW

File: iroha_client/src/client.rs [hunk 1]
```diff
@@ -1,16 +1,76 @@
-use iroha::isi::Contract;
+use iroha::prelude::*;
+use std::{io::prelude::*, net::TcpStream};
 
-#[derive(Default)]
-pub struct Client {}
+const QUERY_REQUEST_HEADER: &[u8; 16] = b"GET / HTTP/1.1\r\n";
+const COMMAND_REQUEST_HEADER: &[u8; 25] = b"POST /commands HTTP/1.1\r\n";
+const OK: &[u8; 19] = b"HTTP/1.1 200 OK\r\n\r\n";
+const INTERNAL_ERROR: &[u8; 38] = b"HTTP/1.1 500 Internal Server Error\r\n\r\n";
```
Comment (eadventurous) [inline iroha_client/src/client.rs:7]: Also duplication here.

File: iroha/src/torii.rs [hunk 2]
```diff
@@ -1,37 +1,44 @@
...
     net::TcpListener,
+    sync::Arc,
+    thread,
     time::{Duration, Instant},
 };
 
 const QUERY_REQUEST_HEADER: &[u8; 16] = b"GET / HTTP/1.1\r\n";
 const COMMAND_REQUEST_HEADER: &[u8; 25] = b"POST /commands HTTP/1.1\r\n";
 const _PUT: &[u8; 16] = b"PUT / HTTP/1.1\r\n";
 const OK: &[u8; 19] = b"HTTP/1.1 200 OK\r\n\r\n";
```
[the remark this answers is not in the document]
Comment (humb1t) [reply iroha/src/torii.rs:13]: @eadventurous your task will be to extract all TCP logic into separate crate

[the remark this answers is not in the document]
Comment (humb1t) [reply iroha/src/torii.rs:13]: https://soramitsu.atlassian.net/browse/RH2-51

File: iroha/src/asset.rs [hunk 3]
```diff
@@ -1,6 +1,7 @@
 use crate::prelude::*;
+use parity_scale_codec::{Decode, Encode};
 
-#[derive(Clone, Debug)]
+#[derive(Clone, Debug, PartialEq, Eq, Ord, PartialOrd, Encode, Decode)]
```
[the remark this answers is not in the document]
Comment (takemiyamakoto) [reply iroha/src/asset.rs:4]: I don't see a need for ordering, except perhaps in queries where returning in alphabetical order is probably polite.

[the remark this answers is not in the document]
Comment (humb1t) [reply iroha/src/asset.rs:4]: I think I made it for BTreeMap, but it works fine without them

**Pull request merged**